dcos_launch/terraform.py

```diff
@@ -0,0 +1,283 @@
+            'private_agents': private_agents_ips,
+            'public_agents': _convert_to_describe_format(_get_ips('Public Agent Public IPs', info))}
+
+        master_elb_address = _get_ips('Master ELB Public IP', info=info)
+        public_agent_elb_address = _get_ips('Public Agent ELB Public IP', info=info)
+
+        if master_elb_address:
+            description.update({'Master ELB Public IP': master_elb_address[0]})
+        if public_agent_elb_address:
+            description.update({'Public Agent ELB Public IP': public_agent_elb_address[0]})
+
+        return description
+
+    def _key_helper_common(self, private_key: bytes):
+        self.config['ssh_private_key'] = private_key.decode('utf-8')
+        with open(self.default_priv_key_path, 'wb') as f:
+            f.write(private_key)
+        # setting private key file permissions to: read and write only, by file owner only
+        os.chmod(self.default_priv_key_path, 0o600)
+        # for launching integration tests
+        self.config['ssh_private_key_filename'] = self.default_priv_key_path
+        # for dcos installation
+        self.config['terraform_config']['ssh_private_key_filename'] = self.default_priv_key_path
+
+    def test(self, args: list, env_dict: dict, test_host: str=None, test_port: int=22, details: dict=None) -> int:
+        # TODO only reason this exists is because private IPs are not yet returned from describe(), which are required
+        # by the parent test() function
+        """ Connects to master host with SSH and then run the internal integration test
+
+        Args:
+            args: a list of args that will follow the py.test command
+            env_dict: the env to use during the test
+        """
+        if args is None:
+            args = list()
+        if self.config['ssh_private_key'] == util.NO_TEST_FLAG or 'ssh_user' not in self.config:
+            raise util.LauncherError('MissingInput', 'DC/OS Launch is missing sufficient SSH info to run tests!')
+        if details is None:
+            details = self.describe()
+        # check for any environment variables that contain spaces
+        env_dict = {e: "'{}'".format(env_dict[e]) if ' ' in env_dict[e] else env_dict[e] for e in env_dict}
+        env_string = ' '.join(['{}={}'.format(e, env_dict[e]) for e in env_dict])
+        arg_string = ' '.join(args)
+        # To support 1.8.9-EE, try using the dcos-integration-test-ee folder if possible
+        pytest_cmd = """ "source /opt/mesosphere/environment.export &&
+cd `find /opt/mesosphere/active/ -name dcos-integration-test* | sort | tail -n 1` &&
+{env} py.test {args}" """.format(env=env_string, args=arg_string)
+        log.info('Running integration test...')
+        if test_host is None:
+            test_host = details['masters'][0]['public_ip']
+        if ':' in test_host:
+            test_host, test_port = test_host.split(':')
+        env_dict['DCOS_DNS_ADDRESS'] = 'http://' + test_host
```

Comment:
There are few places that use this style, can this be `"http://{}".format(test_host)` instead?